From ba0434f825afa0ef5f2b33b92b466c1d445f7b2c Mon Sep 17 00:00:00 2001
From: Chris Lamb <lamby@debian.org>
Date: Thu, 28 Jul 2016 00:30:56 -0400
Subject: [PATCH] Pass --modification-date to xorriso from SOURCE_DATE_EPOCH if
 set

Signed-off-by: Chris Lamb <lamby@debian.org>
---
 scripts/build/binary_iso | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/build/binary_iso b/scripts/build/binary_iso
index e81a13a..d6b59e3 100755
--- a/scripts/build/binary_iso
+++ b/scripts/build/binary_iso
@@ -107,6 +107,12 @@ then
 	XORRISO_OPTIONS="${XORRISO_OPTIONS} -V \"${LB_ISO_VOLUME}\""
 fi
 
+# Set an explicit modification date
+if [ -n "${SOURCE_DATE_EPOCH}" ]
+then
+	XORRISO_OPTIONS="${XORRISO_OPTIONS} --modification-date=$(date --utc --date="@${SOURCE_DATE_EPOCH}" +%Y%m%d%H%m%S00)"
+fi
+
 # Handle xorriso architecture specific options
 case "${LB_BOOTLOADER}" in
 	grub)
-- 
2.8.1

