--- libisofs-1.3.2.orig/libisofs/util.c
+++ libisofs-1.3.2/libisofs/util.c
@@ -1401,6 +1401,33 @@ uint32_t iso_read_bb(const uint8_t *buf,
     return v1;
 }
 
+time_t iso_current_time()
+{
+    char *source_date_epoch, *endptr;
+    time_t now = time(NULL);
+    unsigned long long epoch;
+
+    source_date_epoch = getenv("SOURCE_DATE_EPOCH");
+
+    if (!source_date_epoch)
+        return now; 
+
+    errno = 0; 
+    epoch = strtoull(source_date_epoch, &endptr, 10); 
+
+    if ((errno == ERANGE && (epoch == ULLONG_MAX || epoch == 0))
+            || (errno != 0 && epoch == 0))
+        return now; 
+    if (endptr == source_date_epoch)
+        return now; 
+    if (*endptr != '\0')
+        return now; 
+    if (epoch > ULONG_MAX)
+        return now; 
+
+    return (time_t)epoch;
+}
+
 void iso_datetime_7(unsigned char *buf, time_t t, int always_gmt)
 {
     static int tzsetup = 0;
--- libisofs-1.3.2.orig/libisofs/util.h
+++ libisofs-1.3.2/libisofs/util.h
@@ -244,6 +244,8 @@ uint32_t iso_read_msb(const uint8_t *buf
  */
 uint32_t iso_read_bb(const uint8_t *buf, int bytes, int *error);
 
+time_t iso_current_time();
+
 /** 
  * Records the date/time into a 7 byte buffer (ECMA-119, 9.1.5)
  * 
