From 5058a394992debb7bbb2479a97d6c111d32acff1 Mon Sep 17 00:00:00 2001
From: Chris Lamb <lamby@debian.org>
Date: Sun, 17 Jul 2016 10:20:01 +0200
Subject: [PATCH 2/2] Pass SOURCE_DATE_EPOCH to chroot.

Signed-off-by: Chris Lamb <lamby@debian.org>
---
 functions/chroot.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/functions/chroot.sh b/functions/chroot.sh
index 23c5046..88c5466 100755
--- a/functions/chroot.sh
+++ b/functions/chroot.sh
@@ -26,6 +26,12 @@ Chroot ()
 		fi
 	done
 
+	# Only pass SOURCE_DATE_EPOCH if its already set
+	if [ "${SOURCE_DATE_EPOCH:-}" != "" ]
+	then
+		ENV="${ENV} SOURCE_DATE_EPOCH=${SOURCE_DATE_EPOCH}"
+	fi
+
 	${_LINUX32} chroot "${CHROOT}" /usr/bin/env -i HOME="/root" PATH="/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin" TERM="${TERM}" DEBIAN_FRONTEND="${LB_DEBCONF_FRONTEND}" DEBIAN_PRIORITY="${LB_DEBCONF_PRIORITY}" DEBCONF_NONINTERACTIVE_SEEN="true" DEBCONF_NOWARNINGS="true" ${ENV} ${COMMANDS}
 
 	return "${?}"
-- 
2.8.1

