--- libisoburn-1.3.2.orig/xorriso/misc_funct.c
+++ libisoburn-1.3.2/xorriso/misc_funct.c
@@ -22,6 +22,8 @@
 #include <sys/time.h>
 #include <time.h>
 #include <sys/utsname.h>
+#include <errno.h>
+#include <limits.h>
 
 
 #include "sfile.h"
@@ -1300,3 +1302,29 @@ int Xorriso__to_upper(char *in, char *ou
  return(in[i] == 0);
 }
 
+time_t Xorriso__current_time()
+{
+    char *source_date_epoch, *endptr;
+    time_t now = time(NULL);
+    unsigned long long epoch;
+
+    source_date_epoch = getenv("SOURCE_DATE_EPOCH");
+
+    if (!source_date_epoch)
+        return now; 
+
+    errno = 0; 
+    epoch = strtoull(source_date_epoch, &endptr, 10); 
+
+    if ((errno == ERANGE && (epoch == ULLONG_MAX || epoch == 0))
+            || (errno != 0 && epoch == 0))
+        return now; 
+    if (endptr == source_date_epoch)
+        return now; 
+    if (*endptr != '\0')
+        return now; 
+    if (epoch > ULONG_MAX)
+        return now; 
+
+    return (time_t)epoch;
+}
--- libisoburn-1.3.2.orig/xorriso/misc_funct.h
+++ libisoburn-1.3.2/xorriso/misc_funct.h
@@ -95,5 +95,7 @@ char *Xorriso__hide_mode_text(int hide_m
 */
 int Xorriso__to_upper(char *in, char *out, int out_size, int flag);
 
+time_t Xorriso__current_time();
+
 #endif /* ! Xorriso_pvt_misc_includeD */
 
