--- libisoburn-1.3.2.orig/libisoburn/isoburn.c
+++ libisoburn-1.3.2/libisoburn/isoburn.c
@@ -536,6 +536,11 @@ int isoburn_prepare_disc_aux(struct burn
                         opts->vol_creation_time, opts->vol_modification_time,
                         opts->vol_expiration_time, opts->vol_effective_time,
                         opts->vol_uuid);
+ if (getenv("SOURCE_DATE_EPOCH")) {
+   iso_write_opts_set_always_gmt(wopts, 1);
+   iso_write_opts_set_default_timestamp(wopts, Xorriso__current_time());
+   iso_write_opts_set_replace_timestamps(wopts, 2);
+ }
  iso_write_opts_attach_jte(wopts, opts->libjte_handle);
  iso_write_opts_set_hfsp_serial_number(wopts, opts->hfsp_serial_number);
 
