--- libisoburn-1.3.2.orig/xorriso/write_run.c
+++ libisoburn-1.3.2/xorriso/write_run.c
@@ -945,7 +945,7 @@ int Xorriso_write_session(struct XorrisO
                              (gid_t) xorriso->global_gid);
  isoburn_igopt_set_out_charset(sopts, out_cs);
  isoburn_igopt_set_fifo_size(sopts, xorriso->fs * 2048);
- Ftimetxt(time(NULL), xorriso->scdbackup_tag_time, 8);
+ Ftimetxt(Xorriso__current_time(), xorriso->scdbackup_tag_time, 8);
  isoburn_igopt_set_scdbackup_tag(sopts, xorriso->scdbackup_tag_name,
                                  xorriso->scdbackup_tag_time,
                                  xorriso->scdbackup_tag_written);
