--- libisoburn-1.3.2.orig/xorriso/misc_funct.c
+++ libisoburn-1.3.2/xorriso/misc_funct.c
@@ -463,6 +463,7 @@ int Decode_ecma119_format(struct tm *erg
 /* 2010040711405800 */
 {
  int i, l, num, utc= 1;
+ time_t normalized;
 
  memset(erg, 0, sizeof(*erg));
  erg->tm_isdst= -1;
@@ -499,6 +500,8 @@ int Decode_ecma119_format(struct tm *erg
  erg->tm_sec= 10*(text[12]-'0')+text[13]-'0';
  if(erg->tm_sec > 59)
    return(0);
+ normalized = mktime(erg);
+ erg = gmtime(&normalized);
  return(1 + !utc);
 }
 
